### DISCUSSION TITLE
artillery vulnerability issue (Prototype Pollution in async CVE-2021-43138)

### DISCUSSION THREAD

**Discussion by aldousalvarez** [artilleryio/artillery#1477] [Artillery]
We are currently doing security scan for all our containers and one vulnerability issue (Prototype Pollution in async) that came up is CVE-2021-43138. As per investigation when we tried to upgrade artillery so that async would be upgraded to a compatible version, package "async": "^1.5.2" was not part of the updated package due to "^". Is there a way we could upgrade it to v2.6.4 so that the vulnerability issues would be resolved and is there any roadmap on updating it as well?

**Comment by hassy** [comment]
hi @aldousalvarez, we're on it! the prototype pollution issue itself is of no concern in Artillery specifically, but we do understand scanning results can be a blocker for some teams.

**Reply by eliezercazares** [reply]
totally off topic, but I was struggling for days with ack and response in socketio engine, I just switched to socketio engine v3 package and it's working very well. maybe this should be highlighted somewhere. thanks for your framework it's really cool.

**Reply by trfridle** [reply]
> hi @aldousalvarez, we're on it! the prototype pollution issue itself is of no concern in Artillery specifically, but we do understand scanning results can be a blocker for some teams.

@hassy  
Are there any updates on this?  While I understand that the issue may not be directly exploitable in the implementation as used in Artillery, it is causing compliance issues and builds to fail.

On May 2nd, SNYK submitted a PR: https://github.com/artilleryio/artillery/pull/1434 that was then reverted later that day: https://github.com/artilleryio/artillery/pull/1432

We are now going on 2 months, and we are running out of "allowance" time on our compliance policies for exceptions.  Even an ETA would be helpful at this point that can be used to justify the continued exemptions.

**Reply by hassy** [reply]
@trfridle thanks for the extra context! Just pushed a fix for the issue which blocked us from upgrading to the latest version of `async` originally.

https://github.com/artilleryio/artillery/pull/1434

Will go out in the next release of the CLI this week.

**Reply by trfridle** [reply]
@hassy Thank you very much for the update. Great news on the fix coming soon!

**Comment by hassy** [comment]
Fixed in v2.0.0-20!

**Comment by aldousalvarez** [comment]
@hassy Thank you so much for the help and for this update.
